docs/source/reference/types.rst

```diff
@@ -102,6 +102,17 @@ the beginning or the end of the index specification::
    >>> numba.float32[::1, :, :]
    array(float32, 3d, F)
 
+This style of type declaration is supported within Numba :term:`JIT` compiled
```

Comment:
This links to docs that make a distinction between JIT compilation and AOT compilation, which I think is misleading in this context. Would it be better if it just didn't mention "JIT" and said "Numba-compiled functions" instead?